haste/windowed_conformal_model/conformal_interestingness_model.py

```diff
@@ -91,27 +95,41 @@ def interestingness(self,
             print('interestingness=0 for image from second field: {}' % metadata['original_filename'], flush=True)
             return {'interestingness': 0}
 
-        # TODO: check to see if we at end of window
-
         if timestamp % WINDOW_SIZE == 0:
             # At the end of the window.
 
-            # TODO: if so, query mongoDB for all historic features for this substream
-            #course_features = self.all_course_features_for_substream(..)
+            course_features = self.all_course_features_for_substream(mongo_collection=mongo_collection,
+                                                                     substream_id=substream_id)
+            # ([correlation], [sum_of_intensities], [x_time])
 
-            # Compute features on the entire timeseries for that well.
-            # TODO: which image features do we want to use?
-            #time_series_features = time_series_features()
+            # BB: ** I'm confused here - how do we do the windowing here? **
 
+            timestamps = course_features[2]
+            timestamps_ndarray = numpy.ndarray(shape=(len(timestamps)),
+                                               dtype=int,
+                                               buffer=numpy.array(timestamps))
 
+            # TODO: which image features do we want to use?
+            sum_of_intensities = course_features[1]
+            sum_of_intensities_ndarray = numpy.ndarray(shape=(len(sum_of_intensities)),
+                                                       dtype=int,
+                                                       buffer=numpy.array(sum_of_intensities))
 
-            # TODO: query the core model (pass in whatever it needs, also window size)
+            # Compute features on the entire timeseries for that well.
+            sum_of_intensities_ts_features = time_series_features(sum_of_intensities_ndarray,
+                                                                  timestamps_ndarray,
+                                                                  timestamp)
+            # = (mean,sd,d1,d2)
 
-            # TODO: interpret the conformal prediction - make a binary decision 1 or 0 on interestingness.
-            # based on epsilon
+            # TODO: query the core model (pass in whatever it needs, also window size)
+            # BB: again, confused about the windowing?!
+            pval = 0.1  # interestingness(all_features, features_for_new_image, all_y):
```

Comment:
confused here.

Reply:
interestingness will return two p-values. One for the 0 category (not interesting) and one for the 1 category (interesting).